samples/net/mqtt_publisher/README.rst

```diff
@@ -97,11 +97,30 @@ of this sample application (i.e., :zephyr_file:`samples/net/mqtt_publisher`) and
    :goals: build flash
    :compact:
 
+If the board is connected directly to the Linux host computer through LAN,
+configure the interface. Open the `zephyr/net-tools
+<https://github.com/zephyrproject-rtos/net-tools>`_ repository and run:
+
+.. code-block:: console
+
+	$ ./net-setup.sh --iface eth0
```

Comment:
The `net-setup.sh` is used with qemu and native-sim boards when those are run in host, and the script setups the interface (usually named zeth) to allow connectivity between the simulated board and the host.
The proposed text gives different impression and is confusing, and IMHO needs some clarifications.

Reply:
Thanks for the quick look @jukkar. I understand that it might be intended for the qemu emulation, but when hooking up the board ethernet (for me a `nucleo_h723zg`) directly to my host computer through a LAN cable, it doesn't work, unless the routing is configured for the network interface. As I noticed that this script does pretty much the same - thought it'd be useful to simply link to that. 

As I ran into the issue when running the sample - I can image other people do as well?

Reply:
I agree with Jukka here, recommending to use `native-sim` scripts in such setup seems more of a hacky workaround.
It looks like some IP misconfiguration between host/board (I assume you use static IP addresses given you connect the board directly to the host?).

Reply:
> As I noticed that this script does pretty much the same - thought it'd be useful to simply link to that.

Ok, understood. So you are just using the script to setup things in host side even when not using qemu/native-sim. I never thought the net-setup script might be used this way. It is "nice" hack / workaround that worked for this use case. I would like to avoid putting this information to the documentation even if it works, as the amount of confusion among user could be high.

Reply:
> I assume you use static IP addresses given you connect the board directly to the host?

Yes exactly. The sample set's the micro-controller's IP to `192.0.2.1` and expects the broker at `192.0.2.2` by default.

> So you are just using the script to setup things in host side even when not using qemu/native-sim

Yes, indeed. Probably the whole TAP interface isn't required, just setting up the routing is enough.

>  I would like to avoid putting this information to the documentation even if it works, as the amount of confusion among user could be high.

Yes agree. Do you think that adding this information to the documentation is useful? I wonder what is intended with this sample configuration - as following it doesn't seem to work. Either for qemu one would have to configure the interface, similar when having an actual board, or am I missing something?

---

What if I add:
```sh
# Change this to the interface to which the LAN cable is connected
export IFACE=enp100s0u2u3
```

```sh
IPV4_ADDR_1="192.0.2.2/24"
IPV4_ROUTE_1="192.0.2.0/24"

sudo ip address add $IPV4_ADDR_1 dev $IFACE
sudo ip route add $IPV4_ROUTE_1 dev $IFACE
sudo ip link set dev $IFACE up
```

You can run `sudo ip addr flush dev $IFACE` to remove the interface (undo the step above).

Reply:
> > I assume you use static IP addresses given you connect the board directly to the host?
> 
> Yes exactly. The sample set's the micro-controller's IP to `192.0.2.1` and expects the broker at `192.0.2.2` by default.

Still, please remember those are just defaults we use in simulated environment. I would actually expect to deal with the IP addresses the other way - not by reconfiguring the host, to match the sample defautl values, but rather update the sample project file, to match the network enviroment they work in. So update the broker IP address to the IP address of the host that runs mosquitto.

And this is exactly what's described (vaguely) in the sample docs - update the config file to match your network setup:
https://docs.zephyrproject.org/latest/samples/net/mqtt_publisher/README.html#build-and-running

Reply:
> but rather update the sample project file, to match the network enviroment they work in

This is the way to go in the live environment. But in the documentation, we should use the IP addresses that are meant for documentation like 192.0.2.0/24 address space. Perhaps this should be emphasized in the documentation i.e., if user wants to use static IP, the addresses should match his/hers local network setup.